metau-capstone/StudyViewController.m

```diff
@@ -66,6 +50,60 @@ - (void)viewDidLoad {
     self.horizontalFlip = CATransform3DMakeRotation(M_PI, 0, 1, 0);
 }
 
+- (void) loadFlashcard {
+    if (self.counter < self.arrayOfCards.count) {
+        Flashcard *card = self.arrayOfCards[self.counter];
+        //BACK SIDE
+        self.back = [[CALayer alloc] init];
+        self.back.frame = CGRectMake(0, 0, 300, 180);
+        self.back.backgroundColor = [[UIColor blackColor] CGColor];
+        self.back.position = CGPointMake(self.view.center.x, self.view.center.y - 50);
+        
+        // add text label to the flashcard
+        CATextLayer *backLabel = [[CATextLayer alloc] init];
+        [backLabel setFont:@"Helvetica-Bold"];
+        [backLabel setFontSize:20];
+        [backLabel setString:card.backText];
+        [backLabel setAlignmentMode:kCAAlignmentCenter];
+        [backLabel setForegroundColor:[[UIColor whiteColor] CGColor]];
+        [backLabel setFrame:CGRectMake(0, 0, 300, 180)];
+        [self.back addSublayer:backLabel];
+        self.back.transform = CATransform3DMakeRotation(M_PI, 0, -1, 0);
+        [self.view.layer addSublayer:self.back];
+        
+        // FRONT SIDE
+        self.front = [[CALayer alloc] init];
+        self.front.frame = CGRectMake(0, 0, 300, 180);
+        self.front.backgroundColor = [[UIColor whiteColor] CGColor];
+        self.front.position = CGPointMake(self.view.center.x, self.view.center.y - 50);
+        
+        // add text label to the flashcard
+        CATextLayer *label = [[CATextLayer alloc] init];
+        [label setFont:@"Helvetica-Bold"];
+        [label setFontSize:20];
+        [label setString:card.frontText];
+        [label setAlignmentMode:kCAAlignmentCenter];
+        [label setForegroundColor:[[UIColor blackColor] CGColor]];
+        [label setFrame:CGRectMake(0, 0, 300, 180)];
+        [self.front addSublayer:label];
```

Comment:
Same with these, we can likely pull a lot of this repeated stuff out into some sort of helper that allows us to easier keep styling consistent, avoid repetition and clean this up :)

Reply:
Oo okay yes! Gotcha, that makes sense. I made refactored so now the layers are created once, all the styling is done once, and the load flashcard method only changes the text on the label and does the flip animation (if necessary).